### ISSUE TITLE
Inference showing no segmentation mask

### ISSUE THREAD

**Issue by Davidyao99** [hkchengrex/XMem#41]
hi, great project! I am following the inference instructions and running the eval.py file. When I look at the output folder, it seems like all the segmentation masks are blank (all 0s) except the first frame. 

![image](https://user-images.githubusercontent.com/52120948/197605607-c474063d-3761-4795-add7-b432ec23059a.png)

Any idea why? I have downloaded the respective data (DAVIS 2016) and pretrained models. I am running using this command:
`python eval.py --output ./output/d16 --dataset D16`

Thank you!

**Comment by hkchengrex** [comment]
This has happened to other people as well: https://github.com/hkchengrex/STCN/issues/9
but I have not been able to reproduce this so I am not sure why.

**Comment by Davidyao99** [comment]
Unfortunately, their solution does not resolve the issue. In my case, the model is not working for any dataset. May I ask what is the exact Cuda and torch library versions that you are using?

**Comment by hkchengrex** [comment]
I tried PyTorch 1.11/1.12 and CUDA 10.3/11.6. I guess you can also try disabling AMP (`--benchmark`)

**Comment by Davidyao99** [comment]
Thank you so much for your response. I reinstalled all of my libraries and somehow it worked! ;))

**Comment by bowen-upenn** [comment]
We encountered the same problem but were able to resolve it successfully. If you are using binary masks in your dataset, it is important to ensure that the masks under Annotations/ only consist of binary values.

In inference/data/video_reader.py, we made modifications to the following code and it worked.
```
if load_mask and path.exists(gt_path):
            mask = Image.open(gt_path).convert('L')
            mask = np.array(mask, dtype=np.uint8)
            # the mask must be binary in our dataset
            mask[mask < 128] = 0
            mask[mask >= 128] = 255
            mask = Image.fromarray(mask).convert('P')
            mask = np.array(mask, dtype=np.uint8)
```
